### PR TITLE
fix bugs preventing Machine from functioning

### PR DESCRIPTION
**Pull request by rmudambi**
### Description

- *Category*: bugfix
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5372

### Changes and notes
Fix non-functional Machine by:
- creating and initializing the state column
- calling the transition function on time-step
- 

### Testing
Tests pass
Fixed broken state-machine tests that were incorrectly passing
Added new tests for initialization

### CODE REVIEW

File: src/vivarium/framework/state_machine.py [hunk 1]
```diff
@@ -191,6 +196,16 @@ class State(Component):
     # Properties #
     ##############
 
+    @property
+    def configuration_defaults(self) -> Dict[str, Any]:
```
Comment (stevebachmeier) [inline src/vivarium/framework/state_machine.py:200]: nit: use `dict`

File: src/vivarium/framework/state_machine.py [hunk 2]
```diff
@@ -199,12 +214,18 @@ def model(self) -> str:
     # Lifecycle methods #
     #####################
 
-    def __init__(self, state_id: str, allow_self_transition: bool = False):
+    def __init__(
+        self,
+        state_id: str,
+        allow_self_transition: bool = False,
+        initialization_weights: Callable[[Builder], LookupTableData] = default_initializer,
```
Comment (stevebachmeier) [inline src/vivarium/framework/state_machine.py:221]: nit: I think `callable` is a built-in now?

File: tests/framework/components/test_component.py [hunk 3]
```diff
@@ -169,9 +169,7 @@ def test_component_initializer_is_registered_and_called_if_defined():
     component = ColumnCreator()
     simulation = InteractiveContext(components=[component])
     population = simulation.get_population()
-    expected_pop_view = pd.DataFrame(
-        {column: 9 for column in component.columns_created}, index=population.index
-    )
+    expected_pop_view = component.get_initial_state(population.index)
```
Comment (stevebachmeier) [inline tests/framework/components/test_component.py:172]: This feels like it's getting a little too close to using the thing you're trying to test.

File: tests/framework/test_state_machine.py [hunk 4]
```diff
@@ -35,38 +23,120 @@ def test_initialize_allowing_self_transition():
...
+        {"population": {"population_size": 10000}, "randomness": {"key_columns": []}}
+    )
+    state_a = State("a", initialization_weights=lambda _: 0.2)
+    state_b = State("b", initialization_weights=lambda _: 0.8)
+    machine = Machine("state", states=[state_a, state_b])
+    simulation = InteractiveContext(components=[machine], configuration=base_config)
+
+    state = simulation.get_population()["state"]
+    assert np.all(simulation.get_population().state != "start")
+    assert round((state == "a").mean(), 1) == 0.2
```
Comment (stevebachmeier) [inline tests/framework/test_state_machine.py:47]: Any reason not to use `np.isclose`? 0.24 would rown down to 0.2 but is an unacceptable 16% different.

**Review by stevebachmeier — APPROVED** [review]

File: src/vivarium/framework/state_machine.py (hunk 2 above)
Comment (rmudambi) [reply src/vivarium/framework/state_machine.py:221]: `callable(foo)` is a function that returns True if `foo` is callable. It's not a something that can be used in typing.

File: tests/framework/test_state_machine.py (hunk 4 above)
Comment (rmudambi) [reply tests/framework/test_state_machine.py:47]: Mostly laziness (and it allows for smaller populations. The other tests in this module use `round`. We should really use the FuzzyChecker.

**Review by albrja — APPROVED** [review]

**Pull request merged**